lib/MockKubeResourceMeta.js

```diff
@@ -42,7 +42,7 @@ module.exports = class MockKubeResourceMeta {
   async get(name, namespace) {
     const ref = JSON.parse(this.uri({ name, namespace }));
     const get = await this.kubeGetResource(ref);
-    if (namespace == 'razeedeploy') {
+    if (namespace != 'basedownloadcontrollertest') {
       return get;
     } else if (get) {
       return { statusCode: 200, body: get };
```

Comment:
What is the use case for wrapping the resource like this? is it just for a test?

Reply:
yes i'm reusing this mock for tests

Reply:
Something about this doesn't feel quite right. I would expect the behavior always to be the same. Does the test check for a status code?

Would it be enough to return something or not? if it returns something it was found, if it doesn't, nothing was found. 
Then We wouldn't need divergent behavior for a test.

What do you think?

Reply:
Or maybe `FetchEnvs` should be updated to understand `{statusCode, body}`

which ever behavior happens in the real world, I think this mock class should emulate.

Reply:
its not the test that checks for a status code, its the existing basecontroller code that expects a status code when calling krm.get, which is part of what i am testing. the real krm is able to handle these different behaviors

Reply:
Do you have a link to where that is? i'm trying to get a better picture of what is going on

Reply:
we've determined that the response is different because of the resolveWithFullResponse option. i have updated the mock to resemble this behavior https://github.com/razee-io/razeedeploy-core/blob/master/lib/BaseController.js#L588